### ISSUE TITLE
Cinnamon segfaults when using Wine

### ISSUE THREAD

**Issue by pgkos** [linuxmint/cinnamon-menus#16]
Cinnamon repeatedly segfaults when running Wine-related apps, e.g. winecfg, or even winetricks in quiet mode (console only, no gui).

Output of `coredumpctl dump`:
```
           PID: 535 (cinnamon)
           UID: 1000 (pgkos)
           GID: 1000 (pgkos)
        Signal: 11 (SEGV)
     Timestamp: Wed 2017-04-26 11:06:07 CEST (6h ago)
  Command Line: cinnamon --replace
    Executable: /usr/bin/cinnamon
 Control Group: /user.slice/user-1000.slice/session-c1.scope
          Unit: session-c1.scope
         Slice: user-1000.slice
       Session: c1
     Owner UID: 1000 (pgkos)
      Hostname: pgkos-arch
       Message: Process 535 (cinnamon) of user 1000 dumped core.
                
                Stack trace of thread 535:
                #0  0x00000000b710a884 g_desktop_app_info_get_is_hidden (libgio-2.0.so.0)
                #1  0x00000000b669af5b desktop_entry_get_hidden (libcinnamon-menu-3.so.0)
                #2  0x00000000b669f182 process_layout (libcinnamon-menu-3.so.0)
                #3  0x00000000b669f425 process_layout (libcinnamon-menu-3.so.0)
                #4  0x00000000b66a3087 gmenu_tree_build_from_layout (libcinnamon-menu-3.so.0)
                #5  0x00000000b7662e23 n/a (libcinnamon.so)
                #6  0x00000000b6fcfc3b g_closure_invoke (libgobject-2.0.so.0)
                #7  0x00000000b6fe201e n/a (libgobject-2.0.so.0)
                #8  0x00000000b6feabb6 g_signal_emit_valist (libgobject-2.0.so.0)
                #9  0x00000000b6feaed5 g_signal_emit (libgobject-2.0.so.0)
                #10 0x00000000b669edf0 gmenu_tree_invoke_monitors (libcinnamon-menu-3.so.0)
                #11 0x00000000b66a404e menu_layout_invoke_monitors (libcinnamon-menu-3.so.0)
                #12 0x00000000b6edd37b g_main_context_dispatch (libglib-2.0.so.0)
                #13 0x00000000b6edd769 n/a (libglib-2.0.so.0)
                #14 0x00000000b6eddb19 g_main_loop_run (libglib-2.0.so.0)
                #15 0x00000000b758b8e5 meta_run (libmuffin.so.0)
                #16 0x0000000008049266 n/a (cinnamon)
                #17 0x00000000b6ccf366 __libc_start_main (libc.so.6)
                #18 0x0000000008049345 n/a (cinnamon)
```

Arch Linux 32-bit
Cinnamon 3.2.8
Wine Staging 2.6

**Comment by pgkos** [comment]
Can be reliably reproduced by running:
`winetricks -q dotnet461`

**Comment by clefebvre** [comment]
I can't reproduce this issue. Can you explain how you get there, i.e. please detail how you set up wine, dotnet etc...

**Comment by clefebvre** [comment]
I pushed some commits from upstream, please test this again to let us know if it's still happening.

**Comment by clefebvre** [comment]
If it's still happening on master, please also test the mimeinfo.cache branch.

**Comment by clefebvre** [comment]
We believe this is fixed in 3.8.
